bambou/nurest_object.py

```diff
@@ -361,26 +361,54 @@ def validate(self):
                                                           'remote_name': attribute.remote_name}
                     continue
 
-            if attribute.min_length and len(value) < attribute.min_length:
-                self._attribute_errors[local_name] = {'title': 'Invalid lenght',
-                                                      'description': 'Attribute %s minimum size should be %s but is %s' % (attribute.remote_name, attribute.min_length, len(value)),
+            if attribute.min_length is not None and len(value) < attribute.min_length:
+                self._attribute_errors[local_name] = {'title': 'Invalid length',
+                                                      'description': 'Attribute %s minimum length should be %s but is %s' % (attribute.remote_name, attribute.min_length, len(value)),
                                                       'remote_name': attribute.remote_name}
                 continue
 
-            if attribute.max_length and len(value) > attribute.max_length:
-                self._attribute_errors[local_name] = {'title': 'Invalid lenght',
-                                                      'description': 'Attribute %s maximum size should be %s but is %s' % (attribute.remote_name, attribute.max_length, len(value)),
+            if attribute.max_length is not None and len(value) > attribute.max_length:
+                self._attribute_errors[local_name] = {'title': 'Invalid length',
+                                                      'description': 'Attribute %s maximum length should be %s but is %s' % (attribute.remote_name, attribute.max_length, len(value)),
                                                       'remote_name': attribute.remote_name}
                 continue
 
-            if attribute.choices and value not in attribute.choices:
-                self._attribute_errors[local_name] = {'title': 'Invalid input',
-                                                      'description': 'Invalid input',
-                                                      'remote_name': attribute.remote_name}
-                continue
+            if attribute.attribute_type == list:
+                valid = True
+                for item in value:
+                    if valid is True:
+                        valid = self._validate_value(local_name, attribute, item)
+            else:
+                self._validate_value(local_name, attribute, value)
 
         return self.is_valid()
 
+    def _validate_value(self, local_name, attribute, value):
```

Comment:
For attribute list we can have as a subtype any of the types used: string, number, enums so individual values should be validated as these types. I would create individual functions for each validation: _validate_min/max_length, _validate_min/max, _validate_type and use them above and for lists

Reply:
There is only one definition for min/max_length.  Does this constrain the length of the list or the length of each item?

Reply:
This is a good point! At this time we probably don't necessarily enforce the min/max and string length for the lists however we should enforce the subtype!